.github/workflows/trivy.yml

```diff
@@ -0,0 +1,87 @@
+---
+#
+# This workflow runs Trivy on a Docker image
+# It can pull the image from a container registry
+# or download a tar file.  The latter is used
+# to check a container image prior to publishing
+# to the registry.
+
+name: Run Trivy on a Docker image and push results to GitHub
+
+on:
+  workflow_call:
+    inputs:
+      NOTEBOOK_TYPE:
+        required: true
+        type: string
+      SOURCE_TYPE: # 'tar' or 'image'
+        required: true
+        type: string
+      TARFILE_NAME: # only used if SOURCE_TYPE=='tar'
+        required: false
+        type: string
+      IMAGE_NAME:
+        required: true
+        type: string
+      EXIT_CODE:
+        required: false
+        type: number
+
+env:
+  sarif_file_name: trivy-results-${{ inputs.NOTEBOOK_TYPE  }}.sarif
+
+jobs:
+  trivy:
+    name: Trivy-${{ inputs.NOTEBOOK_TYPE  }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download tar file
+        id: tar-download
+        uses: actions/download-artifact@v4
+        if: ${{ inputs.SOURCE_TYPE == 'tar' }}
+        with:
+          name: ${{ inputs.TARFILE_NAME }}
+          path: /tmp
+
+      - name: load docker image from tar file
+        if: ${{ inputs.SOURCE_TYPE == 'tar' }}
+        run: cat ${{ steps.tar-download.outputs.download-path
+          }}/${{ inputs.TARFILE_NAME
+          }} | docker import - ${{ inputs.IMAGE_NAME }}
+
+      - name: Run Trivy vulnerability scanner for any major issues
+        uses: aquasecurity/trivy-action@0.24.0
+        id: trivy
+        with:
+          image-ref: ${{ inputs.IMAGE_NAME }}
+          ignore-unfixed: true # skip vul'ns for which there is no fix
+          # list files to skip, each with a justification
+          skip-files: |
+            /etc/ssl/private/ssl-cert-snakeoil.key
+          # ssl-cert-snakeoil.key is req'd by the ssl package.
+          severity: 'CRITICAL,HIGH'
```

Comment:
i would suggest parameterizing this

Reply:
@zaro0508 I think you are suggesting passing the `severity` parameter as an input to the workflow.  (The same could be done for the other parameters, like `ignore-unfixed` and `skip-files`.) But  I specifically do not want to do that:  I factored out the Trivy workflow so that when it is called by either the build-push workflow or the periodic workflow it would work in exactly the same way.   I didn't want to have to keep two copies of the same parameters in sync' in the two different workflows.  So I do not want to pass those parameters in from outside.  The way it is written guarantees the scan is always done the same way.

Reply:
i'm only suggesting this parameter because i think we might want to configure it differently for different environments.  You could pass in with a default value which would set them all the same but would also allow overriding.

Reply:
By "environment" do you mean dev vs. prod?  If so, are thinking that we might bypass vul'n scanning in dev'?  I think of this build step as being similar to the 'test' stage: I wouldn't want to modify or bypass the test stage in dev', nor would I want to bypass the vul'n scan stage.

If I misunderstand you, then please explain further.